File: gluecodium/src/test/resources/smoke/structs/output/android/com/example/smoke/StructsWithMethodsInterface.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
import com.example.NativeBase;
public final class StructsWithMethodsInterface extends NativeBase {
    public static final class Vector3 {
        public double x;
        public double y;
        public double z;
        public Vector3(@NonNull final String input) {
            Vector3 _other = create(input);
            this.x = _other.x;
            this.y = _other.y;
            this.z = _other.z;
        }
        public Vector3(@NonNull final StructsWithMethodsInterface.Vector3 other) throws ValidationException {
            Vector3 _other = create(other);
            this.x = _other.x;
            this.y = _other.y;
            this.z = _other.z;
        }
        public native double distanceTo(@NonNull final StructsWithMethodsInterface.Vector3 other);
        @NonNull
        public native StructsWithMethodsInterface.Vector3 add(@NonNull final StructsWithMethodsInterface.Vector3 other);
        public static native boolean validate(final double x, final double y, final double z);
        private static native StructsWithMethodsInterface.Vector3 create(@NonNull final String input);
        private static native StructsWithMethodsInterface.Vector3 create(@NonNull final StructsWithMethodsInterface.Vector3 other) throws ValidationException;
    }
    public static final class StructWithStaticMethodsOnly {
        public static native void doStuff();
    }
    /**
     * For internal use only.
     * @exclude
     */
    protected StructsWithMethodsInterface(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
}

File: gluecodium/src/test/resources/smoke/structs/output/android-kotlin/com/example/smoke/StructsWithMethodsInterface.kt
/*

 *
 */

@file:JvmName("StructsWithMethodsInterfaceExtensions")


package com.example.smoke

import com.example.NativeBase

class StructsWithMethodsInterface : NativeBase {

    class Vector3 {
        @JvmField var x: Double
        @JvmField var y: Double
        @JvmField var z: Double





       constructor(input: String) {
            val _other = create(input)
            this.x = _other.x
            this.y = _other.y
            this.z = _other.z
        }


    @Throws(ValidationUtils.ValidationException::class)
       constructor(other: StructsWithMethodsInterface.Vector3) {
            val _other = create(other)
            this.x = _other.x
            this.y = _other.y
            this.z = _other.z
        }






        external fun distanceTo(other: StructsWithMethodsInterface.Vector3) : Double


        external fun add(other: StructsWithMethodsInterface.Vector3) : StructsWithMethodsInterface.Vector3


        companion object {


            @JvmStatic external fun validate(x: Double, y: Double, z: Double) : Boolean

            @JvmStatic private external fun create(input: String) : Vector3
            @Throws(ValidationUtils.ValidationException::class)
            @JvmStatic private external fun create(other: StructsWithMethodsInterface.Vector3) : Vector3
        }
    }

    class StructWithStaticMethodsOnly {








        companion object {


            @JvmStatic external fun doStuff() : Unit
        }
    }



    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}







    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}


